apps/timer.py

```diff
@@ -88,7 +105,28 @@ def tick(self, ticks):
 
     def touch(self, event):
         """Notify the application of a touchscreen touch event."""
-        if self.state == _RINGING:
+        if self.state == _SELECT:
+            for btn_idx, btn in enumerate(self.quick_btns):
+                if btn.touch(event):
+                    if btn_idx == _CUSTOM_BUTTON:
+                        self._set_delay()
+                        self.state = _STOPPED
+                        self._draw()
+                    else:
+                        if btn_idx == _FIVE_BUTTON:
+                            self._set_delay(5)
+                        elif btn_idx == _TEN_BUTTON:
+                            self._set_delay(10)
+                        elif btn_idx == _FIFTEEN_BUTTON:
+                            self._set_delay(15)
+                        elif btn_idx == _THIRTY_BUTTON:
+                            self._set_delay(30)
+                        elif btn_idx == _FORTY_FIVE_BUTTON:
+                            self._set_delay(45)
+                        elif btn_idx == _SIXTY_BUTTON:
+                            self._set_delay(60)
```

Comment:
I wonder if using a dictionary and giving each button an index equal to the time would be more storage/memory-efficient. Something like
```python
_CUSTOM_BUTTON = const(0)
...
self.quick_btns = {
    5: widgets.Button(20, 50, 60, 40, '5'),
    10: widgets.Button(90, 50, 60, 40, '10'),
    ...
    _CUSTOM_BUTTON: widgets.Button(20, 150, 200, 40, 'Custom')
}
...
for btn_idx in keys(self.quick_btns):
    if self.quick_btns[btn_idx].touch(event):
        if btn_idx == _CUSTOM_BUTTON:
            ...
        else:
            self._set_delay(btn_idx)
```

Reply:
Ya, I made the choice to be as efficient in RAM as I could at cost of ROM. This if else block with constants uses very little RAM when frozen into the firmware. In my experiments the dictionary is pretty RAM heavy. The buttons only exist when the app is opened and are released when it is no longer visible. 

For what its worth this only added a couple KB to my firmware and I have been using it daily for the better part of a year and have not encountered an out of memory error.